mycroft/enclosure/hardware/MycroftLed/led_xmos_usb.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from mycroft.enclosure.hardware.MycroftLed.MycroftLed import MycroftLed
+
+class Led(MycroftLed):
+    """note: we try to minimize the number of led writes 
+       by not writing the same value to the same led"""
+
+    real_num_leds = 12      # physical
+    num_leds = 10           # logical
+    device_addr = 74
+    num_bytes  = 4
+    fifty_zeros = "0 " * 50
+    black = (0,0,0)
+    vfctrl = "/home/pi/mycroft-core/mycroft/enclosure/hardware/util/vfctrl_usb"
+
+    def __init__(self):
+        self.shadow_leds = list((self.black,) * self.num_leds)
+        self.buffered_leds = list((self.black,) * self.num_leds)
+        self.capabilities = {
+                        "num_leds":10,
+                        "led_colors":"MycroftPalette",
+                        "reserved_leds":[10,11],
+                    }
+
+    def get_capabilities(self):
+        return self.capabilities
+
+    def _set_led(self,pixel,colors):
+        """physically set the led and update its shadow"""
```

Comment:
Because the sj201 leds I have with the at tiny are painfully slow this code attempts to minimize the number of led updates by shadowing the data which is believed to be in the at tiny's memory (ie leds) and only sending out requests for leds which have actually changed.